test/LetConst/funcDeclConflict.js

```diff
@@ -0,0 +1,149 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+function f1_f(){};
+var f2_v = 123;
+function f3_f(){};
+var f4_v = 123;
+
+let f5_l = 123;
+const f6_c = 123;
+
+// Top level function in script does not conflict with another top level function
+WScript.LoadScript("function f1_f(){};");
+
+// Top level function in script does not conflict with same-named var
+WScript.LoadScript("function f2_v(){};");
+
+// Top level function in eval does not conflict with another top level function
+eval("function f3_f(){};");
+
+// Top level function in eval does not conflict with same-named var
+eval("function f4_v(){};");
+
+// Top level function in script conflicts with top level let
+try { WScript.LoadScript("function f5_l(){};"); } catch (e) { print(e); }

```

Comment:
Instead of printing the error you could compare the message here and say e.message == "Let/Const redeclaration"

Reply:
I had it all using `assert` and discovered that having a function frame between dynamically parsed code and the outer `let` may have effect on whether it is considered a redeclaration or not. That results in false positive/negative results. 

Nesting, and especially in functions have complicated effects on these scenarios, so I decided that it would be more readable if in the end if `assert.throws` is not used here.

I can still use `assert.areEqual` though.
Will change.